base/loading.jl

```diff
@@ -974,14 +974,14 @@ function explicit_manifest_deps_get(project_file::String, where::PkgId, name::St
             entry = entry::Dict{String, Any}
             uuid = get(entry, "uuid", nothing)::Union{String, Nothing}
             uuid === nothing && continue
+            # deps is either a list of names (deps = ["DepA", "DepB"]) or
+            # a table of entries (deps = {"DepA" = "6ea...", "DepB" = "55d..."}
+            deps = get(entry, "deps", nothing)::Union{Vector{String}, Dict{String, Any}, Nothing}
```

Comment:
Is this `deps` lookup fix related? Does it need a test to match?

Reply:
It is partly related since this runs into https://github.com/JuliaLang/julia/issues/55939 due to how the manifest change with the revert. So it does in some sense already has a test.